dix/dixfonts.c

```diff
@@ -92,7 +98,11 @@ static int size_slept_fpes = 0;
 
 static FontPathElementPtr *slept_fpes = (FontPathElementPtr *) 0;
 
+#ifdef XFONT2
+static xfont2_pattern_cache_ptr patternCache;
```

Comment:
this could go into the header too

Reply:
How should I put this in the header?
FontPatternCachePtr, the type I'm ifdef-ing is also available in libXfont2, but it's the wrong type to use.
It is also a typedef, so I can't just #undef it.

Granted, it's an opaque struct pointer, so I can just use a void* or pointer type for it.
It won't break aliasing rules because it isn't dereferenced in the tinyx code.
Instead it's passed as a pointer to the libXfont/libXfont2 functions and dereferenced there
with the correct type.

Should I do that and just put 'static pointer patternCache;' in the source file?

Reply:
in that case, maybe define (or typedef) pattern_cache_ptr to either of the 2 types used, and use that new type instead

Reply:
done